src/App.tsx

```diff
@@ -3,12 +3,35 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  searchHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.currentTarget.value,
+    });
+  };
+
+  sortHandler = () => {
+    const movies = moviesFromServer;
+    const { query } = this.state;
+
+    return (
+      movies.filter(item => item.title.toLowerCase()
```

Comment:
no need for movies variables, just use moviesFromServer
```suggestion
    const { query } = this.state;

    return (
      moviesFromServer.filter(item => item.title.toLowerCase()
```